### ISSUE TITLE
Doesn't go after [*] Waiting for master server to respond with relay server to use...

### ISSUE THREAD

**Issue by itsStar** [dunderhay/CCTV-v380-pro#3]
The poc1_getcreds.py doesn't work stuck on [*] Waiting for master server to respond with relay server to use...

**Comment by dunderhay** [comment]
What camera model do you have?

Have you tried enabling some of the debugging print statements or adding some debugging in to see what data is being sent / received by the script*?

**Comment by itsStar** [comment]
@dunderhay Thanks for your reply.
Here is the log.
[*] Advertise a new camera on the network to master server with device ID: *********.
Received response from server: b'\xd4\x00\x00\x00vg\xe5\xfb\x00\x00\x00\x00\x00\x00\x00\x00'
[*] Waiting for master server to respond with relay server to use...
And it stuck here.
Camera model is IMOU V380-K6 4MM

**Comment by dunderhay** [comment]
Hmm okay so seems like that response is different from what I got (the packet starting with `d4`):

![image](https://user-images.githubusercontent.com/27876907/123745421-a01f7200-d8f3-11eb-8962-64083f9ec91c.png)

Perhaps your camera might be sending something slightly different to what mine was and so the server is sending back an error instead of attempting to authenticate to the advertised device.

The best way to debug from here would be to intercept the packets between your camera, client (mobile app) and the server. I did this by setting up a raspberry pi (RPI) connected to my test network and then connecting my phone and the IP camera to the RPI. Run tcpdump on the RPI and open the capture file in wireshark. You want to see what a normal flow / conversation between the camera & backend server looks like and then try replicate that with the script.

Hope that helps!
